lib/Agrammon/OutputFormatter/Text.pm6

```diff
@@ -5,28 +5,34 @@ use Agrammon::Outputs;
 sub output-as-text(
     Agrammon::Model $model,
     Agrammon::Outputs $outputs,
-    Str $language, Str $reports-selected,
-    Bool $include-filters, Bool :$all-filters = False, --> Str
+    Str $language,
+    @print-set,
+    Bool $include-filters,
+    Bool :$all-filters = False, --> Str
 ) is export {
     my @lines;
-    my @print-set = $reports-selected.split(',') if defined $reports-selected;
     for sorted-kv($outputs.get-outputs-hash) -> $module, $_ {
         my $n = 0;
         my @module-lines;
         my $indent = '    ';
         push @module-lines, $module;
         when Hash {
             for sorted-kv($_) -> $output, $value {
-                my $val = flat-value($value // 'UNDEFINED');
                 my $var-print = $model.output-print($module, $output) ~ ',All';
-                if not defined $reports-selected or $var-print.split(',') ∩ @print-set {
-                    $n++;
-                    my $unit = $model.output-unit($module, $output, $language);
+                next unless $var-print.split(',') ∩ @print-set;
+
+                $n++;
```

Comment:
I'm trying to figure out what `$n` is for here. It seems to be it will always be non-zero if `@module-lines` has any data, and later we do `@lines.append: @module-lines if $n;` but if `@module-lines` is empty this is a no-op.

Reply:
The idea is, that if we have a module, that has no outputs, then the title line (with the module name) should **not**. be added to the output. `@module-lines` is never empty (it always has at least the title or in case of multi-modules also a sutitle), but if there is no output, `$n == 0`.

I refactored it to use a separate array for the titles and now can check `if @module-lines` without the need for `$n`.

I also renamed `@module-lines` to `@output-lines`.